### PR TITLE
[skip ci] MongooseIM-High-level-Architecture

### PR DESCRIPTION
**Pull request by Nyco**
This PR addresses the need to expose our achitecture

### CODE REVIEW

File: doc/user-guide/MongooseIM-High-level-Architecture.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+## Inside MongooseIM
+
+MongooseIM comes with a large set of modules to configure and enable. See 'Extension Modules' for more info.
```
Comment (michalwski) [inline doc/user-guide/MongooseIM-High-level-Architecture.md]: The `'Extension Modules'` should be a link to the relevant page :)

File: doc/user-guide/MongooseIM-High-level-Architecture.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+MongooseIM manages two sets of data: transient for session data management, and persistent for archive and configurations.
+
+## MongooseIM and its ecosystem in a datacenter
+
+### Databases
+
+Please refer to 'Database Backends' doc for more configuration information.
```
Comment (michalwski) [inline doc/user-guide/MongooseIM-High-level-Architecture.md]: Please make a link from `'Database Backends'`

File: doc/user-guide/MongooseIM-High-level-Architecture.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+## Inside MongooseIM
+
+MongooseIM comes with a large set of modules to configure and enable. See 'Extension Modules' for more info.
+
+MongooseIM manages two sets of data: transient for session data management, and persistent for archive and configurations.
+
+## MongooseIM and its ecosystem in a datacenter
+
+### Databases
```
Comment (michalwski) [inline doc/user-guide/MongooseIM-High-level-Architecture.md]: I'm not able to understand what the author had in mind in this section :) Maybe the formatting is wrong or I miss ed sth.

File: doc/user-guide/MongooseIM-High-level-Architecture.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+Both RDBMS (SQL) and NOSQL (Riak KV, Cassandra) 
+
+* LDAP
+
+LDAP will also run on a separate cluster.
+
+### Frontend
+
+Native clients, on platforms such as Android, iOS, Windows, Linux, macOS, will preferrably use a plain XMPP over TCP connections. Web clients cannot use these TCP connections.
```
Comment (michalwski) [inline doc/user-guide/MongooseIM-High-level-Architecture.md]: The sentence `Web clients cannot use these TCP connections.` is probably redundant. Web clients are described in the next paragraph.

File: doc/user-guide/MongooseIM-High-level-Architecture.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
+## Inside MongooseIM
+
+MongooseIM comes with a large set of modules that sysadmins can configure and potentially enable, modifying the behaviour of MongooseIM. See '[Extension Modules](../advanced-configuration/Modules/)' for more info.
```
Comment (michalwski) [inline doc/user-guide/MongooseIM-High-level-Architecture.md]: You missed the `.md` ext in both links :)

File: doc/user-guide/MongooseIM-High-level-Architecture.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+LDAP will also run on a separate cluster.
+
+### Frontend
+
+Native clients, on platforms such as Android, iOS, Windows, Linux, macOS, will preferrably use a plain XMPP over TCP connections. Web clients cannot use these TCP connections.
+
+Web clients will preferrably use XMPP over websockets, or the now less relevant XMPP over BOSH, using long-lived HTTP connections, more and more used as fallback.
+
+Any client could use the client REST API, which is using HTTP request/responses.
```
Comment (michalwski) [inline doc/user-guide/MongooseIM-High-level-Architecture.md]: We don't have it yet :) we can merge this PR only when the functionality is merged.

**Comment by michalwski** [comment]
All in all, the doc looks like a diagram description, but I don't see any diagram in this PR.

**Comment by Nyco** [comment]
Here we go: https://trello.com/c/MPzqySFm/206-10-5-m-make-our-docs-more-appealing-more-graphical
Still ugly, needs some "beautification".
How do you integrate it?

**Comment by michalwski** [comment]
You need to add the file to the repo. Probably it's possible via the Web UI. Far easier from the console though :D

**Comment by michalwski** [comment]
How about services like https://www.lucidchart.com or any other for drawing diagrams? You can find better images there.

**Comment by michalwski** [comment]
Next step - embed the diagram in the doc :)

File: doc/user-guide/MongooseIM-High-level-Architecture.md [hunk 7]
```diff
@@ -46,6 +46,8 @@ A HTTP notification enables the forwarding of events to any external HTTP servic
 
 WombatOAM enables the monitoring and management of MongooseIM clusters, as well as Riak KV, RabbitMQ, and any other Erlang and Elixir based system.
 
+![MongooseIM high-level architecture](MongooseIM_high-level_architecture.png)
```
Comment (michalwski) [inline doc/user-guide/MongooseIM-High-level-Architecture.md]: Would it be better to have the diagram at the top of the page?

Comment (ludwikbukowski) [inline doc/user-guide/MongooseIM-High-level-Architecture.md]: shouldn't be there some connections (e.g lines) between mongooseim cluster and databases?
Im talking about mongooseim architecture diagram

Comment (Dzol) [inline doc/user-guide/MongooseIM-High-level-Architecture.md]: > shouldn't be there some connections (e.g lines) between mongooseim cluster and databases?

That's true, some lines to **_connect the dots_** as it were, might help bring conceptual clarity to the diagram. At the moment we have pieces but it's not clear how they fit together. Perhaps we can line the DBs up in a row alongside the HTTP box too.

**Pull request merged**